config/puma/production.rb

```diff
@@ -27,6 +27,8 @@
 #
 preload_app!
 
+bind ENV.fetch('FOREMAN_BIND', 'tcp://127.0.0.1')
```

Comment:
This is backwards incompatible. `bind` only accepts a URL but if users have a systemd override that's `FOREMAN_BIND=127.0.0.1` (as our installer sets up), this will fail. You're also missing a port here.

I also wonder if rails server also sets up a default bind, so you would end up binding twice.

Reply:
`rails server` does not appear to set up an additional bind:

```
Sep 10 15:38:00 puma.war.example.com foreman[7270]: BINDS: ["unix:///run/foreman.sock"]
```

Yes, this is backwards incompatible without modifying the systemd override which the corresponding installer change is intended to do. What is the situation where this patch would get used and not accompanied by the installer change?

Are you asking for some processing code for FOREMAN_BIND to ensure it ends up in the proper format?

Reply:
> What is the situation where this patch would get used and not accompanied by the installer change?

If a user runs the update without running the installer (which is supported according to https://theforeman.org/manuals/2.1/index.html#3.6Upgrade).

> Are you asking for some processing code for FOREMAN_BIND to ensure it ends up in the proper format?

Yes. Perhaps it should emit a deprecation message in the logs.

Reply:
Btw, I opened https://github.com/puma/puma/pull/2362 as well which would remove the need for duplicate configuration altogether.